drafts/liberty-dev-mode-gradle.adoc

```diff
@@ -0,0 +1,65 @@
+
+If you would like to give dev mode with Gradle a try you can follow the steps in our https://github.com/OpenLiberty/demo-devmode[demo-devmode] sample project.
+
+== Setting up the project
+
+For Gradle projects, dev mode is run through the Liberty Gradle Plugin.  Specify the Liberty Gradle Plugin with version `3.0` or above in your project’s `build.gradle` file.
+[source,xml]
+----
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'io.openliberty.tools:liberty-gradle-plugin:3.0'
+    }
+}
+----
+
+== Starting dev mode
+
+Start dev mode by running the following command in your project: `gradle libertyDev`
+
+== Automatically recompiling and deploying to the running server
+
+Once dev mode is started, source code changes are automatically compiled and deployed to the running server.  You can add new dependencies in your `build.gradle` file to be used for compilation.  Resources and server configuration changes are also picked up.  If any features are added to the `server.xml` file, they are installed automatically.
+
+== Running tests automatically and on demand
+
+To run tests on demand, press Enter in the terminal where dev mode is running.
+
+To run tests automatically after every change, include the `--hotTests` parameter when starting dev mode.  For example: `gradle libertyDev --hotTests`
+
+== Attaching a debugger
+
+At any point during dev mode, you can attach a Java debugger to debug your application.  By default, the debug port is `7777`, but another port can be used and shown in the console output if the default port is not available.
+
+== Stopping dev mode
+
+You can quit dev mode at any time by pressing CTRL+C in the terminal, or type `q` and press Enter.
+
+== VS Code extension
+
+The https://marketplace.visualstudio.com/items?itemName=Open-Liberty.liberty-dev-vscode-ext[Open Liberty Tools VS Code extension] has also been updated to support Gradle projects. This optional extension lets you interact with dev mode entirely within VS Code. You can start Open Liberty in development mode, make changes to your application while the server is up, run tests and view results, and even debug the application without leaving the editor.
+
+image::/img/blog/liberty-dev-vscode-gradle.png[Liberty Dev Dashboard, align="left", width=400]
+
+For more information on the Open Liberty Tools VS Code extension you can read about it in its link:/blog/2019/11/13/liberty-dev-mode-vscode.html[own blog post].
```

Comment:
Needs a comma.  Change link text to be more descriptive on what the link is.
e.g. `For more information on the Open Liberty Tools VS Code extension, you can read about it in the link:/blog/2019/11/13/liberty-dev-mode-vscode.html[Open Liberty dev mode with VS Code blog post].`